environments/qa/modules/oaeqaautomation/manifests/init.pp

```diff
@@ -78,7 +79,7 @@
     $cron_log_file_path = "${log_file_dir}/`date +'\\%Y-\\%m-\\%d-\\%H-\\%M'`.log"
     cron { 'nightly-redeploy':
         ensure      => present,
-        command     => "${scripts_dir}/nightly.sh >> ${cron_log_file_path} 2>> ${cron_log_file_path}",
+        command     => "test -e /var/lock/redeploy || ${scripts_dir}/redeploy.sh >> ${cron_log_file_path} 2>> ${cron_log_file_path}",
```

Comment:
Seems reasonable, I'll adapt the script to check for the lock file itself
rather than only affecting cronjobs.


On 12 May 2015 at 14:13, Simon Gaeremynck <notifications@github.com> wrote:

> In environments/qa/modules/oaeqaautomation/manifests/init.pp
> <https://github.com/oaeproject/puppet-hilary/pull/147#discussion_r30131469>
> :
>
> > @@ -78,7 +79,7 @@
> >      $cron_log_file_path = "${log_file_dir}/`date +'\\%Y-\\%m-\\%d-\\%H-\\%M'`.log"
> >      cron { 'nightly-redeploy':
> >          ensure      => present,
> > -        command     => "${scripts_dir}/nightly.sh >> ${cron_log_file_path} 2>> ${cron_log_file_path}",
> > +        command     => "test -e /var/lock/redeploy || ${scripts_dir}/redeploy.sh >> ${cron_log_file_path} 2>> ${cron_log_file_path}",
>
> Should we add this test to the redeploy script as well?
>
> I've accidentally wiped a QA server by just running
> /opt/scripts/nightly.sh. Maybe it can be overridden by running /opt/scripts/nightly.sh
> --force or something.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oaeproject/puppet-hilary/pull/147/files#r30131469>.
>